src/praisonai-agents/praisonaiagents/runtime/doctor_registry.py

```
@@ -0,0 +1,147 @@
+            if rule.rule_id == rule_id:
+                return rule
+        return None
+    
+    def collect_all_findings(self, config: Dict[str, Any]) -> List[Finding]:
+        """Collect findings from all registered rules."""
+        all_findings = []
+        for rule in self.get_rules():
+            try:
+                findings = rule.collect_findings(config)
+                all_findings.extend(findings)
+            except Exception as e:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)